### PR TITLE
Feat/43/factory.sh config for tracker and skills

### LINKED ISSUE #43: factory.sh config: set the issue tracker and conventions skills
## What to build
A config subcommand so a user sets up a repo for the factory without hand-editing files:
- `factory.sh config tracker github|linear [--team <linear-team-key>]` — stores the issue tracker in an untracked `.factory/config` in the target checkout; linear requires a team key
- `factory.sh config skills "<skill-name>: <when it applies>" [more...]` — writes the entries to `.factory/conventions` in the structured format from #35
- `factory.sh config` with no args prints the current tracker and skills
Both files live under `.factory/` and stay out of source control via the existing `.git/info/exclude` handling. github is the default tracker when no config exists.

## Acceptance criteria
- [ ] `factory.sh config tracker linear --team ABC` then `factory.sh config` shows tracker linear, team ABC; `tracker github` switches back
- [ ] `factory.sh config skills "euc-go: Go services" "euc-sql: migrations"` writes both entries to `.factory/conventions` in the #35 format and `factory.sh config` lists them
- [ ] `.factory/` is excluded from source control in the target checkout
- [ ] Invalid tracker value or linear without a team fails with a usage message
- [ ] README documents the subcommand
- [ ] Demo: a fresh checkout, three config commands, and the printed output showing everything set

## Blocked by
None

### PR DESCRIPTION
**Pull request by michaelnaughton-euc**
factory.sh config sets a checkout up without hand-editing: config tracker github|linear --team <key> stores the tracker in .factory/config, config skills writes .factory/conventions in the #35 line format, and bare config prints both. Both files stay untracked via .git/info/exclude, github is the default tracker, and invalid tracker or linear without a team fails with a usage message. tests/config.sh covers the acceptance criteria end to end.

Closes #43

### CODE REVIEW

File: factory.sh [hunk 1]
```diff
@@ -168,6 +194,72 @@ repo_dir() {
...
+config_dir() {
+  if [[ -n "$REPO" ]]; then
+    repo_dir
+  elif [[ -d "$WORKSPACE/.git" ]]; then
+    printf '%s\n' "$WORKSPACE"
+  else
+    echo "Need --repo <name> or a git checkout" >&2
+    exit 1
+  fi
+}
```
Comment (michaelnaughton-euc) [inline factory.sh:211]: **Bug (verified):** `infer_github` runs before the lane dispatch and fills `REPO` from the cwd's origin remote, so this `elif [[ -d "$WORKSPACE/.git" ]]` fallback is nearly dead and `config` doesn't actually target "the checkout you run it from" (README's claim) — it targets `$WORKSPACE/<origin-repo-name>` via `repo_dir`.

Repro with `FACTORY_WORKSPACE=$WS`, cwd `$WS/widgets-fork` (origin `acme/widgets`):
- `$WS/widgets` exists → `config tracker github` exits 0 and **silently writes `$WS/widgets/.factory/config`** — the wrong checkout, no warning.
- `$WS/widgets` absent → `No git checkout for widgets under $WS`, even though the cwd is a valid checkout.

`tests/config.sh:73` only passes because it leaves `FACTORY_WORKSPACE` unset there, so `WORKSPACE=$PWD` and `repo_dir`'s second fallback fires by coincidence.

Suggest: when `--repo` was not explicitly passed, resolve from the cwd first (`git rev-parse --show-toplevel 2>/dev/null`) before consulting the workspace — and add the `FACTORY_WORKSPACE`-set repro to the test.

File: factory.sh [hunk 2]
```diff
@@ -168,6 +194,72 @@ repo_dir() {
...
+
+factory_exclude() {
+  local dir="$1"
+  [[ -d "$dir/.git" ]] || return 0
+  grep -qxF '.factory/' "$dir/.git/info/exclude" 2>/dev/null && return 0
+  mkdir -p "$dir/.git/info"
+  printf '.factory/\n' >> "$dir/.git/info/exclude"
+}
```
Comment (michaelnaughton-euc) [inline factory.sh:221]: **Corrupts existing excludes (verified):** if `.git/info/exclude` lacks a trailing newline, the append glues onto the last pattern — `node_modules` becomes `node_modules.factory/`, which both un-excludes `node_modules` and leaves a bogus pattern behind; the `grep -qxF` guard then no longer matches and the next run appends `.factory/` again.

Guard before appending, e.g.:
```bash
[[ ! -s "$dir/.git/info/exclude" || "$(tail -c1 "$dir/.git/info/exclude")" == "" ]] || printf '\n' >> "$dir/.git/info/exclude"
```

File: factory.sh [hunk 3]
```diff
@@ -168,6 +194,72 @@ repo_dir() {
...
+
+config_skills() {
+  local dir="$1"
+  shift
+  [[ $# -gt 0 ]] || { echo 'Need entries: factory.sh config skills "<skill-name>: <when it applies>" [more...]' >&2; exit 1; }
+  mkdir -p "$dir/.factory"
+  factory_exclude "$dir"
+  printf '%s\n' "$@" > "$dir/.factory/conventions"
+  config_show "$dir"
+}
```
Comment (michaelnaughton-euc) [inline factory.sh:271]: **Nit — format not enforced:** the ticket promises entries "in the structured format from #35", but any string is accepted, and an argument with an embedded newline splits into several lines (verified: `config skills $'a: one\nnot-an-entry'` writes a bare `not-an-entry` line). A one-line reject — entry must contain `: ` and no newline — keeps the file honest for whatever consumes it later.

File: factory.sh [hunk 4]
```diff
@@ -74,7 +77,30 @@ NEXT_STEPS=""
...
-if [[ "$LANE" == "mem" ]]; then
+CONFIG_CMD=""
+CONFIG_ARGS=()
+TEAM=""
+
+if [[ "$LANE" == "config" ]]; then
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      --repo) REPO="${2:-}"; shift 2 ;;
+      --team) TEAM="${2:-}"; shift 2 ;;
```
Comment (michaelnaughton-euc) [inline factory.sh:89]: **Nit:** `config tracker linear --team` with the value missing exits 1 with no message (`shift 2` on a single remaining arg under `set -e`). Same flaw as the existing parsers, so fine to leave — flagging only because this PR's stated point is friendly usage errors.

**Review by michaelnaughton-euc — COMMENTED** [review]
Factory review lane. Verdict: **request changes** (posted as comment — reviewer account is the PR author, so GitHub blocks a formal REQUEST_CHANGES).

**Blocker:** `config` run from inside a checkout can silently write `.factory/config` into a *sibling* checkout named after the origin remote, or error out despite standing in a valid checkout — see the inline comment on `config_dir`. Verified locally with a repro; `tests/config.sh` misses this because it never sets `FACTORY_WORKSPACE` in the in-checkout case.

**Also worth a decision, not code:**
- Nothing reads `.factory/config` yet — `tracker linear --team ABC` is write-only; every lane still hardcodes `gh`/GitHub URLs. Fine if the consumer is a follow-up, but say so in the PR body and link the follow-up issue.
- Issue #43 acceptance criterion "Demo: a fresh checkout, three config commands, and the printed output" is not in the PR body or comments.

**Good:** the test is real and CI picks it up automatically (`tests/*.sh` glob in tests.yml); replace-not-append skills semantics and the exclude idempotency are covered; arg parsing matches the house style; `ok config` passes locally.

**Comment by michaelnaughton-euc** [comment]
Review fixes pushed in cd7024a (plus a merge of main, which grew a conflict with #39).

- `config_dir` resolves the cwd checkout first via `git rev-parse --show-toplevel` when `--repo` wasn't passed explicitly. `infer_github`'s inferred `REPO` no longer hijacks the target. Explicit `--repo` still wins. Both repros from the review are now in `tests/config.sh` with `FACTORY_WORKSPACE` set: config from `widgets-fork` writes `widgets-fork/.factory/config` and leaves `widgets` untouched.
- `factory_exclude` appends a newline first when the exclude file doesn't end with one. Test seeds `.git/info/exclude` with a bare `node_modules`, checks the pattern survives, and checks `.factory/` isn't appended twice. #39's `conventions_rules` had the same inline append with the same bug, so it now calls `factory_exclude`.
- `config skills` rejects any entry that lacks `": "` or contains a newline, and a rejected call leaves the existing conventions file alone.
- The missing `--team` value nit is left as is, matching the existing parsers, per the review.

On write-only: since #39 merged, the feature/bug/docs lanes read `.factory/conventions` through `conventions_rules`. `.factory/config` (tracker/team) still has no consumer. That wants a follow-up ticket; flagging it for the lead lane rather than filing it from this one.

Demo, fresh checkout:

```
$ factory.sh config tracker linear --team ENG
tracker linear
team ENG
skills: none

$ factory.sh config skills "euc-go: Go services" "euc-sql: migrations and sqlc queries"
tracker linear
team ENG
skills:
euc-go: Go services
euc-sql: migrations and sqlc queries

$ factory.sh config
tracker linear
team ENG
skills:
euc-go: Go services
euc-sql: migrations and sqlc queries

$ tail -1 .git/info/exclude
.factory/
```

File: factory.sh [hunk 5]
```diff
@@ -168,13 +195,86 @@ repo_dir() {
...
+    printf '%s\n' "$WORKSPACE"
+  else
+    echo "Need --repo <name> or a git checkout" >&2
+    exit 1
+  fi
+}
+
+factory_exclude() {
+  local dir="$1" exclude="$1/.git/info/exclude"
+  [[ -d "$dir/.git" ]] || return 0
```
Comment (michaelnaughton-euc) [inline factory.sh]: **Bug (verified): worktree checkouts silently skip the exclude.** In a worktree, `.git` is a file, so this `return 0` fires, `.factory/` never lands in an exclude, and `git status` shows `?? .factory/` — a lane agent running in that worktree can commit it. Violates #43's AC ".factory/ is excluded from source control in the target checkout" and the README sentence this PR adds.

Repro (PR head):
```bash
git -C "$T/main" worktree add .worktrees/wt -b wtb
cd "$T/main/.worktrees/wt" && factory.sh config tracker github
git status --porcelain   # ?? .factory/
```

Let git compute the path instead — less code, handles both layouts (in a worktree it resolves to the shared `info/exclude`, which is what you want):
```bash
factory_exclude() {
  local dir="$1" exclude
  exclude="$(git -C "$dir" rev-parse --git-path info/exclude 2>/dev/null)" || return 0
  [[ "$exclude" == /* ]] || exclude="$dir/$exclude"
  grep -qxF '.factory/' "$exclude" 2>/dev/null && return 0
  mkdir -p "${exclude%/*}"
  [[ ! -s "$exclude" || -z "$(tail -c1 "$exclude")" ]] || printf '\n' >> "$exclude"
  printf '.factory/\n' >> "$exclude"
}
```
`config_dir`'s `[[ -d "$WORKSPACE/.git" ]]` fallback makes the same `.git`-is-a-dir assumption; `git -C "$WORKSPACE" rev-parse --show-toplevel` would cover it too. Worth a worktree case in `tests/config.sh` since it already covers the other exclude edge cases.

File: factory.sh [hunk 6]
```diff
@@ -958,6 +1058,14 @@ case "$LANE" in
   close-linked)
     close_linked_issues
     ;;
+  config)
+    DIR="$(config_dir)"
+    case "$CONFIG_CMD" in
+      tracker) config_tracker "$DIR" "${CONFIG_ARGS[0]:-}" ;;
```
Comment (michaelnaughton-euc) [inline factory.sh]: **Nit:** extra positional args are silently dropped — `config tracker github linear` writes `tracker=github` and ignores `linear`, since only `CONFIG_ARGS[0]` is read. A one-line reject when `${#CONFIG_ARGS[@]} > 1` keeps typos loud, in the same spirit as the usage errors this PR adds elsewhere.

**Review by michaelnaughton-euc — COMMENTED** [review]
**Verdict: request changes** (GitHub blocks a formal request-changes on your own PR, so posting as a comment review).

Re-review after the fix round. The three earlier findings are fixed and test-covered — cwd targeting via `config_dir`, the newline-glue guard in `factory_exclude`, and the skills format reject — verified in the diff, and `tests/config.sh` passes locally (`ok config`). Extracting `factory_exclude` out of `conventions_rules` is a genuine cleanup, and the test file is thorough.

One verified bug remains, so requesting changes: `factory_exclude` silently does nothing in a git worktree checkout, leaving `.factory/` visible to `git status`. That breaks #43's AC (".factory/ is excluded from source control in the target checkout") and the README sentence this PR adds. Repro and a smaller fix in the inline comment.

Non-blocking notes:
- `factory.sh` was already over 1k lines before this PR (1025 → 1133). The config block is self-contained and single-file is the house style, so fine here — but the file is on a trajectory that will want a split eventually.
- Nothing consumes `tracker=`/`team=` outside `config_show` yet. That matches #43's scope; the consumer should be its own ticket (already flagged to the lead).

**Comment by michaelnaughton-euc** [comment]
Re-reviewed PR 46: prior 3 findings verified fixed, tests pass; verdict request changes on a verified worktree exclude no-op plus 1 nit

[PR 46](https://github.com/Kripu77/software-factory/pull/46)

File: factory.sh (hunk 5 above)
Comment (michaelnaughton-euc) [reply factory.sh]: Fixed in f6d0ca1: `factory_exclude` now resolves the path via `git rev-parse --git-path info/exclude`, so worktrees land in the shared exclude. `config_dir` drops the `-d .git` fallback for `git -C "$WORKSPACE" rev-parse --show-toplevel`. Worktree case added to tests/config.sh (asserts clean `git status --porcelain` from inside the worktree).

File: factory.sh (hunk 6 above)
Comment (michaelnaughton-euc) [reply factory.sh]: Fixed in f6d0ca1: `config tracker` with more than one positional arg now exits 1 with a usage message; test added.

File: factory.sh [hunk 7]
```diff
@@ -168,13 +195,85 @@ repo_dir() {
...
+  local dir="$1" tracker="${2:-}"
+  case "$tracker" in
+    github) ;;
+    linear) [[ -n "$TEAM" ]] || { echo "Tracker linear needs --team <linear-team-key>" >&2; exit 1; } ;;
+    *) echo "Tracker must be github or linear" >&2; exit 1 ;;
+  esac
+  mkdir -p "$dir/.factory"
+  factory_exclude "$dir"
+  if [[ "$tracker" == linear ]]; then
+    printf 'tracker=linear\nteam=%s\n' "$TEAM" > "$dir/.factory/config"
```
Comment (michaelnaughton-euc) [inline factory.sh:253]: **Nit (verified):** `TEAM` is written unvalidated, so a value with an embedded newline injects arbitrary `key=value` lines into `.factory/config`. Repro: `config tracker linear --team $'ABC\ntracker=github'` writes

```
tracker=linear
team=ABC
tracker=github
```

and `config` then reports `tracker github` — the opposite of what was asked, with no error. The skills entries already reject newlines; the same one-line check on `$TEAM` (no newline, and arguably no leading `tracker=`-style ambiguity, i.e. no `=`/whitespace) keeps the file honest. Not blocking.

**Review by michaelnaughton-euc — COMMENTED** [review]
**Verdict: approve** (posted as a comment review — reviewer account is the PR author, so GitHub blocks a formal APPROVE).

Third round. Both round-2 findings are verified fixed at f6d0ca1:

- `factory_exclude` resolves the exclude path via `git rev-parse --git-path info/exclude`, so worktree checkouts land in the shared exclude. Verified by the new worktree case in `tests/config.sh`, which asserts a clean `git status --porcelain` from inside the worktree.
- `config tracker` with extra positional args now exits 1 with a usage message instead of silently dropping them; test added.

`tests/config.sh` passes locally (`ok config`) and CI is green. Structurally this round is a net improvement: `conventions_rules` reusing `factory_exclude` deletes the duplicated inline append, and the worktree fix made `config_dir` smaller, not bigger.

One new nit inline (a `--team` value containing a newline injects extra `key=value` lines into `.factory/config`) — same bug class as the skills-entry reject this PR already added, verified with a repro, but not blocking.

Standing non-blocking notes from earlier rounds, not re-litigated: `factory.sh` is at 1135 lines and will want a split eventually; nothing consumes `tracker=`/`team=` yet (consumer ticket flagged to the lead).

**Comment by michaelnaughton-euc** [comment]
Third-round review of PR 46: both round-2 findings verified fixed, tests and CI green; verdict approve with 1 non-blocking nit (newline in --team injects config lines)

[PR 46](https://github.com/Kripu77/software-factory/pull/46)

**Review by Kripu77 — APPROVED** [review]
